File: 2025-1/sesion-de-clase-13 - Solucion/SoftProgPersistencia/DAO/ICrud.cs
using System.Collections.Generic;

namespace PUCP.Edu.Pe.SoftProg.Persistencia.DAO {
    public interface ICrud<T> {
        int Insertar(T modelo);
        bool Modificar(T modelo);
        bool Eliminar(int id);
        T Buscar(int id);
        List<T> Listar();
    }
}

File: 2025-1/sesion-de-clase-23/SoftProg/SoftProgPersistencia/src/main/java/pe/edu/pucp/inf30/softprog/dao/ICrud.java
package pe.edu.pucp.inf30.softprog.dao;

import java.sql.Connection;
import java.util.List;

/**
 *
 * @author eric
 * @param <T>
 */
public interface ICrud<T> {
    int insertar(T modelo);
    int insertar(T modelo, Connection conexion);
    boolean modificar(T modelo);
    boolean modificar(T modelo, Connection conexion);
    boolean eliminar(int id);
    boolean eliminar(int id, Connection conexion);
    T buscar(int id);
    List<T> listar();
}